### PR TITLE
Local Coherence Deflation

### PR DESCRIPTION
**Pull request by rrhodgson**
Here is the proposal for the overhaul of the deflation setup and adding the local coherence (coarse) deflation.
Requires Grid PR [#426](https://github.com/paboyle/Grid/pull/426) to be accepted first.

Includes:
- Eigenpack compressor/decompressor
- Add coarse deflation
- Make source casting its own module
- Combine bathed and non-batched version of modules
- Removed reloading batched deflation
- Add batched mixed precision solver

### CODE REVIEW

File: Hadrons/Modules/MUtilities/EigenPackLCCompress.hpp
Comment (aportelli) [inline Hadrons/Modules/MUtilities/EigenPackLCCompress.hpp]: Fix comment

File: Hadrons/Modules/MUtilities/EigenPackLCDecompress.hpp
Comment (aportelli) [inline Hadrons/Modules/MUtilities/EigenPackLCDecompress.hpp]: Fix comment

**Review by aportelli — CHANGES_REQUESTED** [review]
Thanks this is really great, please address the changes below. I will merge in `develop` once we have more visibility about fixes needed (or not) after Grid's `feature/dirichlet` merge.
- [ ] please address comments in code
- [ ] please use `name_` only for private variables and not function arguments

**Comment by rrhodgson** [comment]
Thanks for the comments. They should be fixed now.
I also changed the `evBatchSize` back to 1. It will run exactly the same but gives more verbose debug output

**Review by aportelli — APPROVED** [review]
Thanks! All good

**Pull request merged**
